### ISSUE TITLE
Flaky/Intermittent FT test timeout in native image: tests.integration.nativeimage.mp1.Mp1Main

### ISSUE THREAD

**Issue by ljnelson** [oracle/helidon#4296]
One example: https://builds.helidon.io/678781B6AE357E68F35B0CF7E3286FDA/logs/34

```
io.helidon.common.Errors$ErrorMessagesException: FATAL: FT Fallback failed. java.lang.RuntimeException: intentional failure at class io.helidon.tests.integration.nativeimage.mp1.Mp1Main
FATAL: FT Retry failed. java.lang.RuntimeException: Failed at class io.helidon.tests.integration.nativeimage.mp1.Mp1Main
FATAL: FT timeout, expected "timeout", actual: "This method should have timed out after 1 second" at class io.helidon.tests.integration.nativeimage.mp1.Mp1Main
	at io.helidon.common.Errors.checkValid(Errors.java:183)
	at io.helidon.tests.integration.nativeimage.mp1.Mp1Main.testBean(Mp1Main.java:258)
	at io.helidon.tests.integration.nativeimage.mp1.Mp1Main.main(Mp1Main.java:103)
Tests finished in 3200 millis
```

**Comment by tjquinno** [comment]
I have seen this in pipeline jobs at least twice in the past few days also. Native only.

**Comment by spericas** [comment]
Quick test on a _heavily loaded_ Mac with GraalVM 21.1.0, no failures.

```
(~/Projects/helidon-spericas-master/tests/integration/native-image/mp-1) >> for i in 1 2 3 4 5 6 7 8 9 10; do target/helidon-tests
-native-image-mp-1; done 2>&1  > test.log
ABean.timeout() has been interrupted after 1004 millis
ABean.timeout() has been interrupted after 1003 millis
ABean.timeout() has been interrupted after 1009 millis
ABean.timeout() has been interrupted after 1003 millis
ABean.timeout() has been interrupted after 1001 millis
ABean.timeout() has been interrupted after 1004 millis
ABean.timeout() has been interrupted after 1000 millis
ABean.timeout() has been interrupted after 1004 millis
ABean.timeout() has been interrupted after 1002 millis
ABean.timeout() has been interrupted after 1000 millis
```
[test.log](https://github.com/oracle/helidon/files/8824811/test.log)

**Comment by spericas** [comment]
I've also tested it with GraalVM 21.3.0 + JDK 11 with same outcome.

**Comment by barchetta** [comment]
We've made some progress and it is looking like this is a regression. The findings:

1. The issue is intermittent
2. Intermittently native-image builds a binary where FaultTolerance does not work. There is no obvious error or issue during the building of the native-image
3. Once a "bad" binary is built, the FT tests fail every time the binary is run
4. I have reproduced this on Oracle Linux 7.7 using graalvm-ce-java11-21.3.0. Two out of six rebuilds resulted in a bad binary
5. Inspecting the good and bad binaries using strings and grep show differences like this:
```
126c126
< io.helidon.microprofile.faulttolerance.CommandFallback$1
---
> io.helidon.microprofile.faulttolerance.CommandFallback$1`=
```
So the "good" binary had entries with the trailing `=` (significance unknown)
6. After reverting helidon-2.x branch to commit 0c3530d6fbce2899111b62223b0636658e58b2d4 the issue appears to go away. I was able to rebuild the binary 8 times without seeing the issue (more iterations are running as I type).
7. So that seems to point the finger at #4216 as introducing a regression
